modules/AlphaWalletFoundation/AlphaWalletFoundation/Tokens/TokenImageFetcher.swift

```diff
@@ -132,14 +138,18 @@ public class TokenImageFetcherImpl: TokenImageFetcher {
 
         switch type {
         case .nativeCryptocurrency:
-            if let img = serverIconImage {
-                return .init(image: .image(.loaded(image: img)), isFinal: true, overlayServerIcon: nil)
+            if let img = iconImageForContractAndChainID(image: serverIconImage, address: contractAddress.eip55String, chainID: server.chainID) {
+                return TokenImage(image: .image(.loaded(image: img)), isFinal: true, overlayServerIcon: nil)
             }
         case .erc20, .erc875, .erc721, .erc721ForTickets, .erc1155:
-            if let img = tokenImage {
-                return .init(image: .image(.loaded(image: img)), isFinal: true, overlayServerIcon: staticOverlayIcon)
+            if let img = iconImageForContractAndChainID(image: tokenImage, address: contractAddress.eip55String, chainID: server.chainID) {
+                return TokenImage(image: .image(.loaded(image: img)), isFinal: true, overlayServerIcon: staticOverlayIcon)
             }
         }
+        
+        if tokenGroupsIdentifier.isSpam(address: contractAddress.eip55String, chainID: server.chainID) {
```

Comment:
Seems like this will never be true because it was checked in the `switch` within the `iconIamgeForContractAndhainID(image:address:chain:)` call, so the `if`-block can be removed.

Reply:
No, it will fail if the tokenImage or serverIconImage does not exist which could happen if the user adds in a custom network

Reply:
Removed.